test.tf

```diff
@@ -0,0 +1,48 @@
+resource "azurerm_cosmosdb_mongo_database" "test" {
```

Comment:
When you are creating a second environment with the same set of infrastructure as prod, you can reduce duplication by using terraform's "workspaces" instead of duplicating your terraform code. You will sometimes want separate prod.tf and test.tf files if there are any significant differences. But everything here can be covered by a variable or two. And sharing the code means you are more confident that they match, and you can update them independently.

You have avoided duplicating the db account which is nice but not really necessary (the cost is based on usage). And sometimes you won't want to reuse the same server. 

The idea is that you have a single set of terraform config, you select your `prod` workspace (`terraform workspace select prod`), run terraform apply and it will update your prod resources. Then you can select your `test` workspace, run terraform apply and it will manage a second set of everything for your test environment.

Reply:
done